extensions/modules/compression/src/main/java/org/exist/xquery/modules/compression/AbstractCompressFunction.java

```diff
@@ -272,7 +272,7 @@ private void compressElement(final OutputStream os, final Element element, final
             final String stripOffset) throws XPathException {
 
         final String ns = element.getNamespaceURI();
-        if(!(element.getNodeName().equals("entry") || (ns != null && !ns.isEmpty()))) {
+        if(!("entry".equals(element.getNodeName()) || (ns != null && !ns.isEmpty()))) {
```

Comment:
Maybe swapping the main comparison would may be even faster...
```suggestion
        if ((ns != null && !ns.isEmpty()) || !"entry".equals(element.getNodeName())) {
```